packages/arb-avm-cpp/data_storage/src/arbstorage.cpp

```diff
@@ -90,3 +90,43 @@ std::unique_ptr<ReadTransaction> ArbStorage::makeReadTransaction() {
 std::unique_ptr<ReadWriteTransaction> ArbStorage::makeReadWriteTransaction() {
     return std::make_unique<ReadWriteTransaction>(datastorage);
 }
+
+rocksdb::Status resetDBExceptInbox(const std::string& dbpath,
+                                   const std::string& arbospath) {
+    ArbCoreConfig coreConfig{};
+
+    {
+        ArbStorage storage{dbpath, coreConfig};
+        {
+            auto tx = storage.makeReadWriteTransaction();
+            saveNextSegmentID(*tx, 0);
+            auto s = tx->commit();
+            if (!s.ok()) {
+                std::cerr << "Error overwriting segment: " << s.ToString()
+                          << std::endl;
+                return s;
+            }
+        }
+    }
+    {
+        DataStorage storage{dbpath};
+        auto s = storage.clearDBExceptInbox();
+        if (!s.ok()) {
+            std::cerr << "Error deleting columns: " << s.ToString()
+                      << std::endl;
+            return s;
+        }
+        storage.closeDb();
+    }
+    {
+        ArbStorage storage{dbpath, coreConfig};
+        auto s = storage.initialize(arbospath);
```

Comment:
I know this was in the old code, but why do we initialize ArbStorage here?

Reply:
removed